src/sentry/utils/snowflake.py

```diff
@@ -72,10 +73,12 @@ def msb_0_ordering(value, width):
 
 
 def generate_snowflake_id(redis_key: str) -> int:
-    segment_values = {
-        VERSION_ID: msb_0_ordering(settings.SNOWFLAKE_VERSION_ID, VERSION_ID.length),
-        REGION_ID: settings.SNOWFLAKE_REGION_ID,
-    }
+    segment_values = {}
+
+    segment_values[VERSION_ID] = msb_0_ordering(settings.SNOWFLAKE_VERSION_ID, VERSION_ID.length)
+
+    local_region = get_region_mapping().get_local_region()
+    segment_values[REGION_ID] = local_region.id if local_region else 0
```

Comment:
Should we lock this `0` in a constant to denote a 'default region'? Just thinking that when snowflake IDs are rolled out into prod, we'd want 0 to map the the main US region, if we haven't set up the region config in SaaS yet.

Reply:
I guess we can think of it as 'default region' or 'unregioned' though, I think we should avoid SnowflakeIDs if we haven't set the region config just yet.

Reply:
As discussed elsewhere, I prefer not to represent a "default region" when what we really mean is "there are no regions in my entire system (which may be a self-hosted monolith)".

> I think we should avoid SnowflakeIDs if we haven't set the region config just yet.

The plan is to have snowflake IDs be used for all organizations and projects, for the sake of parity, regardless of whether the system is multi-region. So it would be normal for future self-hosted systems to be generating snowflake IDs with zeroes where the region IDs would be if they had regions. (CC @maxiuyuan)

In contrast to not wanting a region *object* as a "default value", the snowflake IDs force us to represent both the "region" and "no-region" cases with numbers. Therefore, I agree with reserving 0 for "unregioned" and documenting it in code (and error-checking it in `Region.__init__`) . We'd then count actual regions from 1.

Reply:
Went ahead and did this in b52156c66c75117cab3403d0ce7b860a4d716783. Let me know if anyone feels strongly about the North American region being 0.